csrc/expr_evaluator.cpp

```diff
@@ -194,4 +195,41 @@ void ExpressionEvaluator::print() const {
   std::cout << "--------------------\n\n";
 }
 
+void ExpressionEvaluator::propagateBoundValuesThroughExactMaps(Fusion* fusion) {
```

Comment:
Could we somehow actually reduce the number of entries in the expression evaluator based on this information? Not to worry about now, but was thinking about this lately. Wonder if we could improve expr eval performance by removing lots of entries since they're equivalent.

Reply:
We could have another layer of mappings, where the first layer would consolidate Vals that are known to have the same value. I guess the actual performance would also depend on the efficiency of the underlying `unordered_map` implementation, which may not be the best data structure for ExpressionEvaluator. For example, I think we could use something like `boost::undered_flat_map`, which is known to be much more efficient than `std::unordered_map`. https://www.boost.org/doc/libs/develop/libs/unordered/doc/html/unordered.html#unordered_flat_map

Reply:
Saw another comment below, yes, we could just simplify the original fusion instead.